### PR TITLE
#3774 Ensure vaccine dispensing button is enabled+warn when no stock

### LINKED ISSUE #3774: MenuPage doesn't refresh to enable vaccine dispensing
## Describe the bug

`MenuPage` doesn't re-render unless modules change which is controlled by `ModuleReducer` which should control whether `VaccineDispensing` 

### To reproduce

1. Add a new vaccine item
2. sync

Also happens if you disable/enable the visibility of the vaccine item.

### Expected behaviour

`MenuPage` should show the vaccine dispensing button when a new, visible vaccine is received.

### Proposed Solution

add `usingVaccineDispensing` to Module reducer.

### Version and device info

- App version: 8.0.0-rc1
- Tablet model:
- OS version:

### Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3774 

## Change summary

- Updated a callback deps list to ensure the menu page is updated correctly after a sync (jeez there's a lot of memoization here)
- Added a toast when a user has no vaccine stock for the vaccination button

## Testing

- [ ] Enable vaccine dispensing, sync and see the UI update
- [ ] Have no stock and try to vaccinate- see the toast

### Related areas to think about

N/a

### CODE REVIEW

File: src/database/UIDatabase.js [hunk 1]
```diff
@@ -252,7 +252,7 @@ class UIDatabase {
       case 'PCDEvents':
         return results.filtered('code == "PCD"');
       case 'Vaccine':
-        return results.filtered('isVaccine == true');
+        return results.filtered('isVaccine == true && isVisible == true');
```
Comment (mark-prins) [inline src/database/UIDatabase.js:255]: oh, hehe. yes, that's a good idea.

File: src/selectors/Entities/vaccinePrescription.js [hunk 2]
```diff
@@ -73,3 +73,8 @@ export const selectSelectedVaccinator = state => {
   const { vaccinator } = VaccinePrescriptionState;
   return vaccinator;
 };
+
+export const selectHaveVaccineStock = () =>
+  UIDatabase.objects('Item')
+    .filtered('isVaccine == true')
```
Comment (mark-prins) [inline src/selectors/Entities/vaccinePrescription.js:79]: perhaps should also have `&& isVisible == true` though I don't know how invisible stock will have available packs. I suppose that is possible

**Review by mark-prins — APPROVED** [review]
👍 
I particularly like the toast, nice touch

**Comment by joshxg** [comment]
Nice! bb26a00

**Pull request merged**